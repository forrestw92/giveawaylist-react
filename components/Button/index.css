.button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  height: 40px;
  width: 125px;
  text-align: center;
  dispaly: flex;
  justify-content: center;
  font-size: 18px;
  box-shadow: 0 1px 1px 0 rgb(0, 0, 0);
  transition: box-shadow 0.25s ease;
  align-items: center;
}
.button.login,
.button.register {
  display: flex;
  margin: auto;
  color: white;
  text-decoration: none;
  background: #1d3557;
}
.button.login:hover,
.button.register:hover {
  box-shadow: 0 3px 1px 1px rgba(0, 0, 0, 0.8);
}
