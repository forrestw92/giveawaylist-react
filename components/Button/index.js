import React from "react";
import { string, func } from "prop-types";
import stylesheet from "./index.css";
function Button(props) {
  const { className, label, _onClick, type, href } = props;
  return (
    <React.Fragment>
      {type === "button" ? (
        <button className={`button ${className}`} onClick={e => _onClick(e)}>
          {label}
        </button>
      ) : (
        <a href={href} className={`button ${className}`}>
          {label}
        </a>
      )}
      <style jsx>{stylesheet}</style>
    </React.Fragment>
  );
}

Button.propTypes = {
  _onClick: func.isRequired,
  label: string.isRequired,
  className: string.isRequired,
  type: string.isRequired,
  href: string
};

export default Button;
