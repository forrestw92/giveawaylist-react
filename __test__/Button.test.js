/* eslint-env jest */

import { render } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import Router from "next/router";
import Button from "../components/Button";
const mockedRouter = { push: () => {}, prefetch: () => {} };
Router.router = mockedRouter;

describe("Button Component", () => {
  it("Button Renders", () => {
    const app = render(
      <Button
        className={"register"}
        _onClick={() => {}}
        type={"a"}
        label={"test"}
        href={"/"}
      />
    );
    expect(app.hasClass("register")).toBe(true);
    expect(app.text()).toEqual("test");
    expect(app.attr("href")).toEqual("/");
  });
});

describe("Button Component Snapshot", () => {
  it("Renders", () => {
    const component = renderer.create(
      <Button
        className={"register"}
        _onClick={() => {}}
        type={"a"}
        label={"test"}
        href={"/"}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
