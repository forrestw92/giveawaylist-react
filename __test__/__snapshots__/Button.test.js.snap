// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshot Renders 1`] = `
<a
  className="button register"
  href="/"
>
  test
</a>
`;
